vim-patch:8.0.0708: some tests are old style + 8.0.1326

blueyed opened the pull request:
Problem:    Some tests are old style.
Solution:   Change a few tests from old style to new style. (pschuh,
            closes vim/vim#1813)

https://github.com/vim/vim/commit/292eff0c5aacb8531d65509679b6c29eae8dc22a

Picked this as base for 8.0.1378/8.0.1377 (which touches file from here).
A lot of deleted files were not present though, so please review carefully.

```
Applying patch...
can't find file to patch at input line 15
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|commit 292eff0c5 (tag: v8.0.0708)
|Author: Bram Moolenaar <Bram@vim.org>
|Date:   Tue Jul 11 21:46:28 2017 +0200
|
|    patch 8.0.0708: some tests are old style
|    
|    Problem:    Some tests are old style.
|    Solution:   Change a few tests from old style to new style. (pschuh,
|                closes #1813)
|
|diff --git a/src/nvim/testdir/Make_all.mak b/src/nvim/testdir/Make_all.mak
|index 9a3cafd37..217a29547 100644
|--- a/src/nvim/testdir/Make_all.mak
|+++ b/src/nvim/testdir/Make_all.mak
--------------------------
File to patch: 
Skip this patch? [y] 
Skipping patch.
4 out of 4 hunks ignored
can't find file to patch at input line 64
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/src/nvim/testdir/main.aap b/src/nvim/testdir/main.aap
|index 821aa2d94..93159ba20 100644
|--- a/src/nvim/testdir/main.aap
|+++ b/src/nvim/testdir/main.aap
--------------------------
File to patch: 
Skip this patch? [y] 
Skipping patch.
1 out of 1 hunk ignored
The next patch would delete the file src/nvim/testdir/test23.in,
which does not exist!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
1 out of 1 hunk ignored
The next patch would delete the file src/nvim/testdir/test23.ok,
which does not exist!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
1 out of 1 hunk ignored
patching file src/nvim/testdir/test24.in
Not deleting file src/nvim/testdir/test24.in as content differs from patch
patching file src/nvim/testdir/test24.ok
The next patch would delete the file src/nvim/testdir/test26.in,
which does not exist!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
1 out of 1 hunk ignored
The next patch would delete the file src/nvim/testdir/test26.ok,
which does not exist!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
1 out of 1 hunk ignored
The next patch would delete the file src/nvim/testdir/test67.in,
which does not exist!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
1 out of 1 hunk ignored
The next patch would delete the file src/nvim/testdir/test67.ok,
which does not exist!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
1 out of 1 hunk ignored
The next patch would delete the file src/nvim/testdir/test75.in,
which does not exist!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
1 out of 1 hunk ignored
The next patch would delete the file src/nvim/testdir/test75.ok,
which does not exist!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
1 out of 1 hunk ignored
The next patch would delete the file src/nvim/testdir/test97.in,
which does not exist!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
1 out of 1 hunk ignored
The next patch would delete the file src/nvim/testdir/test97.ok,
which does not exist!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
1 out of 1 hunk ignored
The next patch would delete the file src/nvim/testdir/test_comparators.in,
which does not exist!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
1 out of 1 hunk ignored
The next patch would delete the file src/nvim/testdir/test_comparators.ok,
which does not exist!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
1 out of 1 hunk ignored
The next patch would create the file src/nvim/testdir/test_comparators.vim,
which already exists!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
1 out of 1 hunk ignored
The next patch would create the file src/nvim/testdir/test_escaped_glob.vim,
which already exists!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
1 out of 1 hunk ignored
The next patch would create the file src/nvim/testdir/test_exec_while_if.vim,
which already exists!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
1 out of 1 hunk ignored
The next patch would create the file src/nvim/testdir/test_exists_autocmd.vim,
which already exists!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
1 out of 1 hunk ignored
The next patch would delete the file src/nvim/testdir/test_getcwd.in,
which does not exist!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
1 out of 1 hunk ignored
The next patch would delete the file src/nvim/testdir/test_getcwd.ok,
which does not exist!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
1 out of 1 hunk ignored
The next patch would create the file src/nvim/testdir/test_getcwd.vim,
which already exists!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
1 out of 1 hunk ignored
The next patch would create the file src/nvim/testdir/test_maparg.vim,
which already exists!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
1 out of 1 hunk ignored
The next patch would create the file src/nvim/testdir/test_plus_arg_edit.vim,
which already exists!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
1 out of 1 hunk ignored
The next patch would create the file src/nvim/testdir/test_regex_char_classes.vim,
which already exists!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
1 out of 1 hunk ignored
```

The already existing files are due to running the script before already.

blueyed committed:
vim-patch:8.0.0708: some tests are old style

Problem:    Some tests are old style.
Solution:   Change a few tests from old style to new style. (pschuh,
            closes vim/vim#1813)

https://github.com/vim/vim/commit/292eff0c5aacb8531d65509679b6c29eae8dc22a

justinmk commented on src/nvim/testdir/Makefile:
```diff
@@ -15,12 +15,10 @@ export TMPDIR := $(abspath ../../../Xtest-tmpdir)
            test40.out             \
            test42.out             \
            test48.out             \
-           test49.out             \
```
need to delete the associated files, too.

justinmk commented:
The new tests need to be added to `testdir/Makefile`, see for example: https://github.com/blueyed/neovim/commit/ec02d9aad6b3015b1802476521a65234c70b0574#diff-b3c6ad6680a25a1b42095879e3a87104

If Vim added the tests to `test_alot.vim`, then also add them there.

blueyed commented:
Updated: rebased and added a fixup addressing the comments from @justinmk.

Should the tests in the Makefile get sorted?

justinmk commented on src/nvim/testdir/Makefile:
```diff
@@ -26,7 +25,6 @@ SCRIPTS_DEFAULT = \
 ifneq ($(OS),Windows_NT)
   SCRIPTS_DEFAULTS := $(SCRIPTS_DEFAULT) \
            test17.out             \
-           test49.out             \
```
Was this intentional? As much as I'd _like_ to remove this, it wasn't removed in the vim patch, and is still present on Vim HEAD.

blueyed replied:
Good catch.  Fixed, and sorted the test files.

justinmk commented:
was `test49.vim` removed intentionally? 

> Should the tests in the Makefile get sorted?

They should be alphanumeric order, if that's what you mean.

justinmk commented:
`test49.vim` is stilled deleted in this PR (but not the vim patch)

blueyed commented:
Fixed.

justinmk commented on src/nvim/testdir/Makefile:
```diff
@@ -35,6 +34,9 @@ SCRIPTS ?= $(SCRIPTS_DEFAULT)
 NEW_TESTS ?= \
+	    test_alot.res \
+	    test_alot_latin.res \
+	    test_alot_utf8.res \
```
The "alot" tests are an exception to the "alphanumeric order" rule. If in doubt, just check Vim's order to see the expected order.

justinmk commented on src/nvim/testdir/Makefile:
```diff
@@ -86,21 +94,23 @@ NEW_TESTS ?= \
 	    test_quickfix.res \
 	    test_quotestar.res \
 	    test_recover.res \
+	    test_regex_char_classes.vim \
```
These all need to be `.res`, not `.vim`.

justinmk commented:
Seems like `8.0.1326` should be ported with this as well (and the patches that it depends on).

blueyed committed:
vim-patch:8.0.1326: largefile test fails on CI, glob test on MS-Windows

Problem:    Largefile test fails on CI, glob test on MS-Windows.
Solution:   Remove largefile test from list of all tests. Don't run
            Test_glob() on non-unix systems.  More cleanup. (Yegappan
            Lakshmanan, closes vim/vim#2354)
https://github.com/vim/vim/commit/6e77df2d8555ade4470e566011603ae40f1f0f3a

blueyed commented:
@janlazo 
Can you see if there is still something from with this PR?

blueyed commented:
AFAIK builds are never green lately, and it is hard for me to grasp any actual error out of them.

janlazo commented:
Why is this failing on Travis? I'd expect it to fail on Appveyor because of `globpath` and `noshellslash` and I don't recall earlier Vim patches that fix this.

```
From test_escaped_glob.vim:
Found errors in Test_globpath():
function RunTheTest[35]..Test_globpath line 1: Expected 'sautest/autoload/Test104.vim\nsautest/autoload/footest.vim' but got 'sautest/autoload/footest.vim'
function RunTheTest[35]..Test_globpath line 3: Expected ['sautest/autoload/Test104.vim', 'sautest/autoload/footest.vim'] but got ['sautest/autoload/footest.vim']
```

janlazo commented:
@blueyed Patch 8.0.1447 modifies  src/testdir/test_regex_char_classes.vim

janlazo commented:
8ca8a0da08bbf8f1670490b4c1cf1b39668816b6 migrated Test104, along with the autoload file required for `Test_globpath`. @blueyed, restore the file or fix the test to omit the missing file.

blueyed commented:
@janlazo 
Thanks, adjusted the test to ignore the missing file.

janlazo commented:
It passes on Travis now but check Appveyor anyway because the builds don't fail for any oldtest failures and oldtests assertions assume `set shellslash`.

janlazo commented:
@blueyed Can you rebase from master to resolve the merge conflict?

janlazo commented on src/nvim/testdir/test_escaped_glob.vim:
```diff
@@ -0,0 +1,32 @@
+  call delete('Xxx$')
+endfunction
+
+function Test_globpath()
```
Fails on Appveyor because `globpath` outputs paths with backslash. This passes for Vim because of `set shellslash` but that breaks `shellescape` for cmd.exe. Neovim uses cmd.exe, not sh, for its shell on Windows.

blueyed replied:
But in `SetUp` it does `set shell=doesnotexist` to not use a shell?

Will try the following now:
```diff
modified: src/nvim/testdir/test_escaped_glob.vim
─────────────────────────────────────────────────────────────────────────────────────────────
@ src/nvim/testdir/test_escaped_glob.vim:28 @ function Test_glob()
 endfunction
 
 function Test_globpath()
-  call assert_equal("sautest/autoload/footest.vim",
+  let slash = (!exists('+shellslash') || &shellslash) ? '/' : '\'
+  call assert_equal('sautest'.slash.'autoload'.slash.'footest.vim',
         \ globpath('sautest/autoload', '*.vim'))
-  call assert_equal(['sautest/autoload/footest.vim'],
+  call assert_equal(['sautest'.slash.'autoload'.slash.'footest.vim'],
         \ globpath('sautest/autoload', '*.vim', 0, 1))
 endfunction
```

janlazo replied:
I use `expand` myself to make it less messy when merging patches but that should be fine.

janlazo replied:
@blueyed `set shell` has no effect on the value of `shellslash`.

blueyed replied:
What should be done here then now?

You said:
> This passes for Vim because of set shellslash

Where does Vim use `set shellslash`?  (at least not in the patch/test directly - https://github.com/vim/vim/commit/292eff0c5aacb8531d65509679b6c29eae8dc22a#diff-fa96be7b9f8aeee55448425e4aa092f2)

Do you mean that Vim uses `set shell`?

janlazo reviewed (changes requested):
Convert the forward slash to backslash so the test passes on Windows or temporarily set/unset shellslash.

janlazo commented:
Why did TSAN build fail on oldtests? The new oldtests added in this PR passed on Appveyor now.

justinmk commented:
> Why did TSAN build fail on oldtests?

Good catch, that happens on master though: it started with https://travis-ci.org/neovim/neovim/builds/357894160 , though may be caused by the previous commit (waiting for it to build): https://travis-ci.org/neovim/neovim/builds/357828588

janlazo commented:
Besides merge conflicts, anything else blocking this?

blueyed commented:
Not from my side.

justinmk commented:
would you mind rebasing this?

blueyed commented:
Rebased.
https://travis-ci.org/neovim/neovim/jobs/421109307 needs to be restarted.

janlazo commented:
`test_largefile.vim` runs, right? 8.0.1326 disabled it but our builds still work if it runs?!

blueyed committed:
test_largefile.vim: adjust comment to run it

blueyed commented:
@janlazo 
Do you mean if it can be run?
I've adjusted the comment in the file, and it appears to run then.

janlazo commented:
@blueyed I thought that it's running on the CI builds but, based on the Makefile, it's not. Does it pass if run manually?

blueyed commented:
Yes.
Easy to try yourself though (given ~4gb of RAM).

janlazo commented:
I assume we need one more run of the CI builds to confirm everything works before merging this. Appveyor builds should pass now with the updated cache. The recently merged vim-patches shouldn't affect this PR.

blueyed commented:
Rebased.

blueyed commented:
https://travis-ci.org/neovim/neovim/jobs/426112172 needs to be restarted.
